packages/teleport/src/Discover/DownloadScript/DownloadScript.tsx

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright 2022 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { Text, Box, Indicator } from 'design';
+import { Danger } from 'design/Alert';
+import cfg from 'teleport/config';
+import TextSelectCopy from 'teleport/components/TextSelectCopy';
+import { useDiscoverContext } from '../discoverContextProvider';
+import { AgentStepProps } from '../types';
+import { useDownloadScript } from './useDownloadScript';
+import type { State } from './useDownloadScript';
+import { Header, ActionButtons } from '../Shared';
+
+export default function Container(props: AgentStepProps) {
+  const ctx = useDiscoverContext();
+  const state = useDownloadScript({ ctx, props });
```

Comment:
I don't really feel like having a separate hook is necessary here, if you moved the join token code into this file you'd save a bunch of unnecessary boiler plate.

Reply:
we can refactor later